dali/operators/reader/loader/video/frames_decoder_gpu.cc

```diff
@@ -216,31 +221,91 @@ int FramesDecoderGpu::ProcessPictureDecode(void *user_data, CUVIDPICPARAMS *pict
   return 1;
 }
 
+int FramesDecoderGpu::ProcessPictureDecodeWithoutIndex(
```

Comment:
I think you can extract common part from here and `ProcessPictureDecodeWithIndex`.